fastlane/lanes/release.rb

```diff
@@ -228,20 +228,34 @@ def trigger_buildkite_release_build(branch:, beta:)
 #
 desc 'Verifies that Gutenberg is referenced by release version and not by commit'
 lane :gutenberg_dep_check do
-  require_relative './../../Gutenberg/version'
-
-  case [GUTENBERG_CONFIG[:tag], GUTENBERG_CONFIG[:commit]]
-  when [nil, nil]
-    UI.user_error!('Could not find any Gutenberg version reference.')
-  when [nil, commit = GUTENBERG_CONFIG[:commit]]
-    if UI.confirm("Gutenberg referenced by commit (#{commit}) instead than by tag. Do you want to continue anyway?")
-      UI.message("Gutenberg version: #{commit}")
+  require 'yaml'
+
+  gutenberg_config_path = File.join(PROJECT_ROOT_FOLDER, 'Gutenberg', 'config.yml')
+
+  UI.user_error!("Could not find config YAML at path #{gutenberg_config_path}") unless File.exist?(gutenberg_config_path)
+
+  begin
+    config = YAML.safe_load(File.read(gutenberg_config_path), symbolize_names: true)
+
+    source = config[:ref]
+
+    UI.user_error!('Gutenberg config does not contain expected key :ref') if source.nil?
+
+    case [source[:tag], source[:commit]]
+    when [nil, nil]
+      UI.user_error!('Could not find any Gutenberg version reference.')
+    when [nil, commit = source[:commit]]
+      if UI.confirm("Gutenberg referenced by commit (#{commit}) instead than by tag. Do you want to continue anyway?")
+        UI.message("Gutenberg version: #{commit}")
+      else
+        UI.user_error!('Aborting...')
+      end
     else
-      UI.user_error!('Aborting...')
+      # If a tag is present, the commit value is ignored
+      UI.message("Gutenberg version: #{source[:tag]}")
     end
-  else
-    # If a tag is present, the commit value is ignored
-    UI.message("Gutenberg version: #{GUTENBERG_CONFIG[:tag]}")
+  rescue ArgumentError => e
+    UI.user_error!("Could not parse config YAML. Failed with: #{e.message}")
```

Comment:
What you think reducing the scope of this rescue to just the YAML parsing `YAML.safe_load` part?

Reply:
Good point. Addressed in a7500d0.

I tested it by changing the config to an invalid one (unterminated `""`)

```diff
 # Example:
 #
 #   LOCAL_GUTENBERG=../my-gutenberg-fork bundle exec pod install
-ref:
-  # This should have either a commit or tag key.
-  # If both are set, tag will take precedence.
-  #
-  # We have automation that reads and updates this file.
-  # Leaving commented keys is discouraged, because as the automation would ignore them and the resulting updates would be noisy
-  tag: v1.101.0
-github_org: wordpress-mobile
-repo_name: gutenberg-mobile
+#
+foo: "bar
```
and verified the error reporting:

<img width="1921" alt="Screenshot 2023-08-17 at 12 48 40 pm" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/471fcfad-eda8-4a7b-b7e2-3da0f2000c15">